sdk/src/accounts/webSocketDriftClientAccountSubscriber.ts

```diff
@@ -353,4 +388,48 @@ export class WebSocketDriftClientAccountSubscriber
 		}
 		return this.oracleSubscribers.get(oraclePublicKey.toString()).dataAndSlot;
 	}
+
+	public getOraclePriceDataAndSlotForPerpMarket(
+		marketIndex: number
+	): DataAndSlot<OraclePriceData> | undefined {
+		const perpMarketAccount = this.getMarketAccountAndSlot(marketIndex);
+		const oracle = this.perpOracleMap.get(marketIndex);
+		if (!perpMarketAccount || !oracle) {
+			return undefined;
+		}
+
+		if (!perpMarketAccount.data.amm.oracle.equals(oracle)) {
+			// If the oracle has changed, we need to update the oracle map in background
+			this.addOracle({
+				source: perpMarketAccount.data.amm.oracleSource,
+				publicKey: perpMarketAccount.data.amm.oracle,
+			}).then(() => {
+				this.setPerpOracleMap();
+			});
+		}
+
+		return this.getOraclePriceDataAndSlot(oracle);
+	}
+
+	public getOraclePriceDataAndSlotForSpotMarket(
+		marketIndex: number
+	): DataAndSlot<OraclePriceData> | undefined {
+		const spotMarketAccount = this.getSpotMarketAccountAndSlot(marketIndex);
+		const oracle = this.spotOracleMap.get(marketIndex);
+		if (!spotMarketAccount || !oracle) {
+			return undefined;
+		}
+
+		if (!spotMarketAccount.data.oracle.equals(oracle)) {
+			// If the oracle has changed, we need to update the oracle map in background
+			this.addOracle({
+				source: spotMarketAccount.data.oracleSource,
+				publicKey: spotMarketAccount.data.oracle,
+			}).then(() => {
+				this.setSpotOracleMap();
```

Comment:
probably want to unsubscribe from the old oracle and tear down the old connection here too